bundles/binding/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpGenericBindingProvider.java

```diff
@@ -186,45 +196,100 @@ private void parseBindingConfig(SnmpBindingConfig config, Item item, String bind
+				config.put(IN_BINDING_KEY, newElement);
+			} else { 
+				// not matched, try with version
+				inMatcher = IN_BINDING_PATTERN_VERSION.matcher(bindingConfig);
+				if (!inMatcher.matches()) {
+					inMatcher = IN_BINDING_PATTERN_VERSION_TRANSFORM.matcher(bindingConfig);
+				}
+				if (inMatcher.matches()) {
+					SnmpBindingConfigElement newElement = new SnmpBindingConfigElement();
+					newElement.address = GenericAddress.parse("udp:" + inMatcher.group(1).toString() + "/161");
+					String version = inMatcher.group(2).toString();
+					if (version.equals("v1")) {
+						newElement.snmpVersion = SnmpConstants.version1;
+					} else if (version.equals("v2c")) {
+						newElement.snmpVersion = SnmpConstants.version2c;
+					} else {
+						newElement.snmpVersion = SnmpConstants.version3;
+					}
+					newElement.community = new OctetString(inMatcher.group(3).toString());
+					newElement.oid = new OID(inMatcher.group(4).toString());
+					newElement.refreshInterval = Integer.valueOf(inMatcher.group(5)).intValue();
+					if(inMatcher.groupCount() == 6)
+						newElement.setTransformationRule(inMatcher.group(6));
+					config.put(IN_BINDING_KEY, newElement);              
+				}
+			}
+            
+			Matcher outMatcher = OUT_BINDING_PATTERN.matcher(bindingConfig);
+			if (outMatcher.matches()) {
 				SnmpBindingConfigElement newElement = new SnmpBindingConfigElement();
+				String commandAsString = outMatcher.group(1).toString();
+				newElement.address = GenericAddress.parse("udp:" + outMatcher.group(2).toString() + "/161");
+				newElement.snmpVersion = SnmpConstants.version1;
+				newElement.community = new OctetString(outMatcher.group(3).toString());
+				newElement.oid = new OID(outMatcher.group(4).toString());
+
+				// Only Integer commands accepted at this time.
+				newElement.value = new Integer32(Integer.parseInt(outMatcher.group(5).toString()));
+
+				Command command = TypeParser.parseCommand(item.getAcceptedCommandTypes(), commandAsString);
+				if (command == null) {
+					logger.error("SNMP can't resolve command {} for item {}", commandAsString, item);
+				} else {
+					config.put(command, newElement);
+				}
+			} else {
+				outMatcher = OUT_BINDING_PATTERN_VERSION.matcher(bindingConfig);
 				if (outMatcher.matches()) {
+					SnmpBindingConfigElement newElement = new SnmpBindingConfigElement();
 					String commandAsString = outMatcher.group(1).toString();
 					newElement.address = GenericAddress.parse("udp:" + outMatcher.group(2).toString() + "/161");
-					newElement.community = new OctetString(outMatcher.group(3).toString());
-					newElement.oid = new OID(outMatcher.group(4).toString());
+					String version = inMatcher.group(3).toString();
+					if (version.equals("v1")) {
+						newElement.snmpVersion = SnmpConstants.version1;
+					} else if (version.equals("v2c")) {
+						newElement.snmpVersion = SnmpConstants.version2c;
+					} else {
+						newElement.snmpVersion = SnmpConstants.version3;
```

Comment:
since v1 was the default before, wouldn't it be better to fall down to v1 in this else case?